### PR TITLE
feat: terraform adapted to watcher v2

### PR DESCRIPTION
**Pull request by 0xGorilla**
Necessary modifications for the deploy repository to support the new version of the watcher.

* Documentation for devnets improvements
* CI improvements
* Tags addition and simplified setup
* Addition of redis

### CODE REVIEW

File: config.tf [hunk 1]
```diff
@@ -1,46 +1,30 @@
...
+    { name = "BETTER_UPTIME_API_KEY", value = var.better_uptime_api_key },
+    { name = "CUSTOM_RPC_PROVIDERS", value = var.custom_rpc_providers },
+    # devnet scripts environment variables
+    { name = "TENDERLY_ACCESS_KEY", value = var.tenderly_access_key },
+    { name = "TENDERLY_ACCOUNT_ID", value = var.tenderly_account_id },
+    { name = "TENDERLY_PROJECT_SLUG", value = var.tenderly_project_slug },
   ]
 }
-
-locals {
```
Comment (carlomazzaferro) [inline config.tf:15]: Is the `NXTP_CONFIG` no longer required in the new watcher codebase?

File: config/modules/iam/ecs.tf [hunk 2]
```diff
@@ -19,6 +19,7 @@ data "aws_iam_policy_document" "task_execution_cloudwatch_access" {
     effect = "Allow"
     actions = [
       "logs:*",
+      "secretsmanager:GetSecretValue"
```
Comment (carlomazzaferro) [inline config/modules/iam/ecs.tf:22]: good catch here

File: main.tf [hunk 3]
```diff
@@ -40,18 +42,35 @@ module "watcher" {
   service_security_groups  = flatten([module.network.allow_all_sg, module.network.ecs_task_sg])
   container_env_vars       = local.watcher_env_vars
```
Comment (carlomazzaferro) [inline main.tf:43]: I think this is not used any longer, or rather, those local variables have been removed

**Comment by carlomazzaferro** [comment]
Let's also add a `.terraform-version` file at the top level: https://github.com/tfutils/tfenv#terraform-version-file

So that we are sure to always use the same one. E.g. in connext: https://github.com/connext/monorepo/blob/main/ops/testnet/staging/backend/.terraform-version

**Comment by 0xGorilla** [comment]
> Let's also add a `.terraform-version` file at the top level: https://github.com/tfutils/tfenv#terraform-version-file
> 
> So that we are sure to always use the same one. E.g. in connext: https://github.com/connext/monorepo/blob/main/ops/testnet/staging/backend/.terraform-version

We have one actually, since it hasn't been changed it doesn't appear in the PR: https://github.com/defi-wonderland/connext-watcher-terraform/blob/dev/.terraform-version

File: config.tf (hunk 1 above)
Comment (0xGorilla) [reply config.tf:15]: Nop

File: config/modules/redis/sgs.tf [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  type              = "ingress"
+  from_port         = 6379
+  to_port           = 6379
+  protocol          = "tcp"
+  cidr_blocks       = ["172.17.0.0/16"]
+  security_group_id = aws_security_group.redis.id
+}
+
+
+resource "aws_security_group_rule" "allow-res-to-redis" {
```
[the remark this answers is not in the document]
Comment (0xGorilla) [reply config/modules/redis/sgs.tf]: Fixed in c8c6b4d

**Review by carlomazzaferro — APPROVED** [review]

**Pull request merged**